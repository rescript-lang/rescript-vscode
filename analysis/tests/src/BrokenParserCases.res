// --- BROKEN PARSER CASES ---
// This below demonstrates an issue when what you're completing is the _last_ labelled argument, and there's a unit application after it. The parser wrongly merges the unit argument as the expression of the labelled argument assignment, where is should really let the trailing unit argument be, and set a %rescript.exprhole as the expression of the assignment, just like it normally does.
// let _ = someFn(~isOff=, ())
//                       ^com

// This should parse as a single item tuple when in a pattern?
// switch s { | (t) }
//                ^com

