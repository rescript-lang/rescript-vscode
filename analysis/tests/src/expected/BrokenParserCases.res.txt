Complete src/BrokenParserCases.res 2:25
posCursor:[2:25] posNoWhite:[2:24] Found expr:[2:11->2:30]
Pexp_apply ...[2:11->2:17] (~isOff2:19->2:24=...[2:27->2:29])
Completable: Cargument Value[someFn]($0)
[]

Complete src/BrokenParserCases.res 6:18
posCursor:[6:18] posNoWhite:[6:17] Found expr:[6:3->6:21]
posCursor:[6:18] posNoWhite:[6:17] Found pattern:[6:16->6:19]
Completable: Cpattern Value[s]=t
[]

